Update functions and introduce model sub-types to support non-constant aerodynamic roughnesses.

akshaysridhar opened the pull request:
Currently supports the `CharnockRoughness` model and `ScalarRoughness` Model.

tapios commented:
Let me know when it's ready for review.

tapios reviewed (approved):
LGTM. It would be good if someone (e.g., @dennisYatunin) have a look at the data structures for the intermediate solution values and the surface conditions. I'm not sure they are ideal.

akshaysridhar committed:
Update function signatures to support non-constant aerodynamic
roughnesses (`CharnockRoughness`). Default behaviour using
prescibed aerodynamic roughness supported through the `ScalarRoughness`
type.

Update to use SecantMethod for both stable and unstable bl.

akshaysridhar commented on src/SurfaceFluxes.jl:
```diff
@@ -436,46 +549,68 @@ function compute_richardson_number(
+
+Computed as: z0m = α_charnock * u★² / g, where α_charnock = 0.011.
+"""
+function compute_charnock_roughness(param_set, u★)
```
Yes. An upcoming change generalizes the roughness functions to scalar surface variables, such that temperature, moisture, roughness can all be prescribed using the same interface.
